### PR TITLE
fix: remove the ability to edit archived classrooms

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link
[Hide all edit options on archived classrooms](https://www.notion.so/uwblueprintexecs/Hide-all-edit-options-on-archived-classrooms-89b3524ae0fd417485d91c69981e7d56)

## Implementation description
* Hide archive options on frontend
* Filter out archive options from backend methods

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by carissa-tang — COMMENTED** [review]
i'm guessing this is not done? but i think we need to also restrict update/delete student and updates to test session

and then make those frontend changes

**Comment by jfdoming** [comment]
> i'm guessing this is not done? but i think we need to also restrict update/delete student and updates to test session
> 
> and then make those frontend changes

It should be done. The changes you mentioned I implemented in #550... probably should have made them here instead but that pr is already basically approved so figure I'll just let this pr fill in the gaps instead

**Comment by jfdoming** [comment]
> > i'm guessing this is not done? but i think we need to also restrict update/delete student and updates to test session
> > and then make those frontend changes
> 
> It should be done. The changes you mentioned I implemented in #550... probably should have made them here instead but that pr is already basically approved so figure I'll just let this pr fill in the gaps instead

Realized I did miss a couple of UI options, though

**Comment by jfdoming** [comment]
image
should be good to go now

File: frontend/src/checkFeatureFlag.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const flags = {
-  ENABLE_CLASSROOM_ARCHIVING: () => false,
+  ENABLE_CLASSROOM_ARCHIVING: () => true,
```
Comment (carissa-tang) [inline frontend/src/checkFeatureFlag.ts]: save this change?

**Review by carissa-tang — APPROVED** [review]
we might want to hide the search bar / filter menu for empty states - but can be follow up PR if you want

**Comment by jfdoming** [comment]
> we might want to hide the search bar / filter menu for empty states - but can be follow up PR if you want

oops yep, fixed

**Pull request merged**